src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java

```diff
@@ -204,6 +333,37 @@ public AxisAlignedBB getBoundingBox(final IBlockState state, final IBlockAccess
         }
     }
 
+    /**
+     * @deprecated remove when minecraft invents something better.
+     */
+    @Deprecated
+    @Override
+    public IBlockState getActualState(@NotNull final IBlockState state, @NotNull final IBlockAccess worldIn, @NotNull final BlockPos pos)
+    {
+        return getTapeShape(state, worldIn, pos);
+    }
+
+    /**
+     * Get the step shape of the slab
+     * @param state the state.
+     * @param world the world.
+     * @param position the position.Re
+     * @return the blockState to use.
+     */
+    private static IBlockState getTapeShape(@NotNull final IBlockState state, @NotNull final IBlockAccess world, @NotNull final BlockPos position)
+    {
+        final boolean[] connectors = new boolean[]{world.getBlockState(position.east()).getBlock() instanceof BlockConstructionTape,
+                world.getBlockState(position.west()).getBlock() instanceof BlockConstructionTape,
+                world.getBlockState(position.north()).getBlock() instanceof BlockConstructionTape,
+                world.getBlockState(position.south()).getBlock() instanceof BlockConstructionTape};
+
+        if((connectors[0] && connectors[2]) || (connectors[0] && connectors[3]) || (connectors[1] && connectors[3]) || (connectors[1] && connectors[2]))
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Reduce the number of conditional operators (7) used in the expression (maximum allowed 6). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1067)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 3 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 3 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)